dvc/repo/get.py

```diff
@@ -53,26 +60,24 @@ def get(url, path, out=None, rev=None):
             #
             # Also, we can't use theoretical "move" link type here, because
             # the same cache file might be used a few times in a directory.
+            repo = Repo(tmp_dir)
             repo.cache.local.cache_types = ["reflink", "hardlink", "copy"]
+            output = repo.find_out_by_relpath(path)
+            if not output.use_cache:
+                # Catch this below and go for a plain old fs_copy
+                raise _DoPlainCopy
```

Comment:
This is a very strange flow control, I have never ever seen one like that, but maybe I haven't seen enough python projects, so please correct me if I'm wrong. Regardless, I pretty sure you could simply do something like:
```
try:                                                                    
    repo = Repo(tmp_dir)                                                
    output = repo.find_out_by_relpath(path)                             
    if output.use_cache:                                                
        ... 
        _get_cached(repo, output, out)                                  
        return                                                          
except (NotDvcRepoError, OutputNotFoundError):                          
    pass                                                                
                                                                        
if os.path.isabs(path):                                                 
    raise FileNotFoundError                                             
                                                                        
fs_copy(os.path.join(tmp_dir, path), out)                               
```
and avoid that.

Reply:
Yes, that is way nicer!